### ISSUE TITLE
Dropdownmenu RangeError Exception when using enableFilter:true with enableSearch:true (default)

### ISSUE THREAD

**Issue by GamalOthman** [flutter/flutter#157697]
### Steps to reproduce

1. Setup the `Dropdownmenu` widget as the code sample provided
2. In emulator, click the widget and input the first character of an entry in the menu, then input any different character.
3. For example, if entries in the dropdown menu have entry of label `251`, input `252`, an exception should happen

### Expected results

filtered entries = 0 as there is no matching text.

### Actual results

App halts. RangeError Exception.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MainApp());
}

class MainApp extends StatefulWidget {
  const MainApp({super.key});

  @override
  State<MainApp> createState() => _MainAppState();
}

class _MainAppState extends State<MainApp> {
  TextEditingController dropCtrlr = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: Padding(
              padding: const EdgeInsets.symmetric(horizontal: 10),
              child: InkWell(
                child: DropdownMenu(
                  controller: dropCtrlr,
                  width: 500,
                  menuHeight: 300,
                  requestFocusOnTap: true,
                  enableFilter: true,
                  enableSearch: true,
                  dropdownMenuEntries:
                      [251,365,441].map<DropdownMenuEntry<int>>(
                          (dynamic object) {
                    return DropdownMenuEntry<int>(
                      value: object,
                      label: object.toString(),
                      enabled: true,
                      style: MenuItemButton.styleFrom(
                        foregroundColor: Colors.black,
                      ),
                    );
                  }).toList(),
                ),
              ),
            ),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![image](https://github.com/user-attachments/assets/4d3ec689-3f9f-47aa-84bd-597cfad55600)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
PS C:\Users\Gamal> flutter doctor -v
[✓] Flutter (Channel stable, 3.24.2, on Microsoft Windows [Version 10.0.22631.4317], locale en-US)
    • Flutter version 3.24.2 on channel stable at G:\sdk\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cf269e36d (8 weeks ago), 2024-09-03 14:30:00 -0700
    • Engine revision a6bd3f1de1
    • Dart version 3.5.2
    • DevTools version 2.37.2

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at C:\Users\Gamal\AppData\Local\Android\sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✗] Visual Studio - develop Windows apps
    ✗ Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[✓] Android Studio (version 2024.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)

[✓] VS Code (version 1.92.2)
    • VS Code at C:\Users\Gamal\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.94.0

[✓] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 15 (API 35) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22631.4317]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 129.0.6668.101
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 130.0.2849.52

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by saturngod** [comment]
same issue happen on Flutter 3.24.4 • channel stable .

**Comment by danagbemava-nc** [comment]
This was initially reported in https://github.com/flutter/flutter/issues/151878 & https://github.com/flutter/flutter/issues/154532
 
And both issues have since been fixed. The fix is in the beta and master channels. 

Since this has already been fixed, I'll be closing this per our triage policy. 

If you experience this issue on either the latest beta or master channels, kindly leave a comment below and I will reopen this. 

Thank you

**Comment by GamalOthman** [comment]
I've upgraded Flutter to the latest stable version 3.24.4 and the problem still persists.
I think the issue is solved here [#151969](https://github.com/flutter/flutter/pull/151969), but I see it is merged into master version. I don't know when it will merge to stable version as I can't use other versions for developing apps. So, I don't want to open the issue now, just waiting.
